examples/proof_of_location.masm

```diff
@@ -0,0 +1,714 @@
+    exec.arcsin
+    # => [arcsin_hi, arcsin_lo]
+
+    push.DOUBLE_EARTH_RADIUS_LO.DOUBLE_EARTH_RADIUS_HI exec.i64_mul
+    # => [distance_hi, distance_hi]
+    # Convert to kilometers.
+    push.SCALING_FACTOR_LO.SCALING_FACTOR_HI exec.i64_div
+    # => [distance_km_hi, distance_km_lo]
+  end
+end
+
+# INPUT CONVERSION
+# =================================================================================================
+
+#! Converts the input format to i64s.
+#!
+#! Inputs: [sign_lat, lat, sign_lon, lon]
+#! Outputs: [lat_hi, lat_lo, lon_hi, lon_lo]
+proc.convert_input_format.2
+  repeat.2
+    # Add the zero high bits.
+    push.0 swap
+    # => [sign_lat, lat_hi, lat_lo, sign_lon, lon] (first loop iteration)
+    if.true
+      # Negate if the sign bool is true.
+      exec.i64_neg
+    end
+    exec.u64_swap
+    # => [sign_lon, lon, lat_hi, lat_lo] (first loop iteration)
+  end
+end
+
+# TESTS
+# =================================================================================================
+
+proc.test_i64_neg
+  # negate(negate(5)) = 5
+  push.5.0
+  exec.i64_neg exec.i64_neg
+  push.5.0 exec.u64::eq assert.err=1
+
+  # negate(negate(5000000000)) = 5000000000
+  push.705032704.1
+  exec.i64_neg exec.i64_neg
+  push.705032704.1 exec.u64::eq assert.err=2
+end
+
+proc.test_i64_mul
+  # 5 * -4 = -20
+  push.5.0.0xfffffffc.0xffffffff
+  exec.i64_mul
+  push.0xffffffec.0xffffffff exec.u64::eq assert
+
+  # -5 * -4 = 20
+  push.0xfffffffb.0xffffffff.0xfffffffc.0xffffffff
+  exec.i64_mul
+  push.20.0 exec.u64::eq assert
+
+  # -5 * 4 = -20
+  push.0xfffffffb.0xffffffff.4.0
+  exec.i64_mul
+  push.0xffffffec.0xffffffff exec.u64::eq assert
+
+  # 5 * 4 = 20
+  push.5.0.4.0
+  exec.i64_mul
+  push.20.0 exec.u64::eq assert
+end
+
+proc.test_i64_div
+  # 20 / -4 = -5
+  push.20.0.0xfffffffc.0xffffffff
+  exec.i64_div
+  push.0xfffffffb.0xffffffff exec.u64::eq assert
+
+  # -20 / 4 = -5
+  push.0xffffffec.0xffffffff.4.0
+  exec.i64_div
+  push.0xfffffffb.0xffffffff exec.u64::eq assert
+
+  # -20 / -4 = 5
+  push.0xffffffec.0xffffffff.0xfffffffc.0xffffffff
+  exec.i64_div
+  push.5.0 exec.u64::eq assert
+
+  # 20 / 4 = 5
+  push.20.0.4.0
+  exec.i64_div
+  push.5.0 exec.u64::eq assert
+end
+
+begin
+  # Reverse coordinates from operand stack.
+  movdn.3 movdn.2 swap
+  exec.convert_input_format
+  # => [berlin_lat_hi, berlin_lat_lo, berlin_lon_hi, berlin_lon_lo]
+
+  # Load coordinates from advice stack and reverse.
+  padw adv_loadw movdn.3 movdn.2 swap
+  exec.convert_input_format
+  # => [berlin_lat_hi, berlin_lat_lo, berlin_lon_hi, berlin_lon_lo,
+  #     secret_location_lat_hi, secret_location_lat_lo, secret_location_lon_hi, secret_location_lon_lo]
+
+  exec.distance_between_coords
+  # => [distance_km_hi, distance_km_lo]
+
+  # Prove that the "secret location" from the advice input is less than 15km from the Brandenburger Tor in Berlin.
+  # i.e. that the location is within the Berlin metropolitan area.
+  push.15.0 exec.u64::lt assert
```

Comment:
This is super nice! You could add an error message at the end by 

`assert.err=123`